.github/workflows/nightly-windows.yml

```diff
@@ -0,0 +1,133 @@
+name: nightly-Windows
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 2 * * *'
+
+jobs:
+  nightly-Windows:
+    strategy:
+      fail-fast: false
+      matrix:
+        arch:
+          - cmake: Win32
+            choco_options: '--x86'
+            address_model: 32
+          - cmake: x64
+            choco_options: ''
+            address_model: 64
+        build_type:
+          - Debug
+          - Release
+        shared_libs:
+          - toggle: OFF
+            name: Static
+          - toggle: ON
+            name: Shared
+        with_openssl:
+          - toggle: OFF
+            name: 'noSSL'
+          - toggle: ON
+            name: 'SSL'
+
+    runs-on: windows-latest
+    name: >-
+      Windows
+      (${{ matrix.arch.address_model }}, ${{ matrix.build_type }}, ${{ matrix.shared_libs.name }}, ${{ matrix.with_openssl.name }})
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Install OpenSSL
+        if: matrix.with_openssl.toggle == 'ON'
+        run: |
+          choco install openssl ${{ matrix.arch.choco_options }}
+          Invoke-WebRequest `
+              "https://curl.se/ca/cacert.pem" `
+              -OutFile "C:\cacert.pem"
+
+      - name: Install Boost
+        run: |
+          Invoke-WebRequest `
+              "https://boostorg.jfrog.io/artifactory/main/release/1.75.0/source/boost_1_75_0.tar.gz" `
+              -OutFile "boost_1_75_0.tar.gz"
+          tar xzf boost_1_75_0.tar.gz
+          rm boost_1_75_0.tar.gz
+          cd boost_1_75_0
+          .\bootstrap.bat
+          .\b2 address-model=${{ matrix.arch.address_model }} --with-thread --with-chrono install
+          cd ..
+          Remove-Item boost_1_75_0 -Recurse -Force
+
+      - name: Install Thrift
```

Comment:
vcpkg has thrift, boost and openssl, why not just use vcpkg for dependency management (and use `-DCMAKE_TOOLCHAIN_FILE` option)? is choco more preferable compared to vcpkg? vcpkg repo already has thrift 0.13.0 in the ports.

Reply:
Yes it could be done, but vcpkg is too slow. Installing Boost, for example, took nearly an hour on Github's runners. Chocolatey is faster because it downloads binaries instead of building packages from source. I would install Boost and Thrift from Chocolatey as well if it had the versions we need but it doesn't.

Reply:
i am still confused how it is faster when you now compile boost manually anyway? hence, no help from choco on the boost side. why do you think that manually compiling boost is faster than having boost being compiled using vcpkg (i wonder if it has anything to do with git clone of vcpkg)?

is choco already installed on the GA container, or do we install it? what about conan versus choco? conan has the boost and thrift as up to date versions available and it downloads binaries? choco also may be completely ok but if conan is used you may need to have python on the windows machine but then you would not need the steps to compile thrift and boost.

Reply:
>  i am still confused how it is faster when you now compile boost manually anyway? hence, no help from choco on the boost side. why do you think that manually compiling boost is faster than having boost being compiled using vcpkg (i wonder if it has anything to do with git clone of vcpkg)?

I don't know the exact reason but it was also mentioned here: https://github.com/Microsoft/vcpkg/issues/3724#issuecomment-605609605 . It's slow even if you specify only a few packages of Boost. 

> is choco already installed on the GA container, or do we install it?

It's already installed. GA virtual environments come contain many preinstalled packages. A detailed list for Windows is given [here](https://github.com/actions/virtual-environments/blob/main/images/win/Windows2019-Readme.md). The preinstalled package managers are under[ this section](https://github.com/actions/virtual-environments/blob/main/images/win/Windows2019-Readme.md#package-management). There is vcpkg but no Conan.

>  conan has the boost and thrift as up to date versions available and it downloads binaries

Conan is not guaranteed to download binaries. Sometimes the binary for your OS and architecture isn't cached on the servers and you have to add the `--build` option. I considered Conan when I was implementing macOS and Linux builds but I decided against it as it required too much setup (creating a conan.yml/py file, passing the toolchain file, etc.).

What is the upside of using a package manager for CI jobs? We currently have a working workflow file, and, considering the build time, I think we are at a good state. Building/downloading OpenSSL + Boost + Thrift takes about 6 minutes, which is 10% of the total CI time. Plus, we have more control when we build from source. For example, I am using `-DBUILD_COMPILER=OFF` when building Thrift because we don't need the thrift compiler.

The package management stuff in this PR is actually very similar to what we did for  Linux. We install OpenSSL from package managers (`apt-get` for Ubuntu, `choco` for Windows), and build Boost and Thrift from source. I kind of think of Chocolatey as the equivalent of `apt` on Ubuntu (I may be wrong). It doesn't provide many packages or options but it's fast and reliable.

Reply:
i was just thinking that if it would be even faster than 6mins if you used conan binary packages. if it is also slow or now available, it is not an important problem and we can live with current solution since this is for preparing test environment anyway.

Reply:
As I said, Conan is not guaranteed to have binaries. For example, [in this workflow](https://github.com/yemreinci/hazelcast-cpp-client/runs/2709457575?check_suite_focus=true) I tried installing dependencies and it couldn't find the binary for thrift/0.13.0. I believe problems like these will be likely to occur for other platforms and architectures as well. Although they can be worked around (adding --build parameter if failed etc.) I don't think they are worth the effort.

I have [also ran with](https://github.com/yemreinci/hazelcast-cpp-client/runs/2709506621?check_suite_focus=true) `--build thrift`, and it took 6.5 mins.